service/service_setup.go

```diff
@@ -111,3 +112,11 @@ func (svc *serviceCore) setupObserver(cfg config.Provider) {
 		}
 	}
 }
+
+func (svc *serviceCore) setupAuthClient(cfg config.Provider) {
```

Comment:
This doesn't seem to be called anywhere in this PR. Is that intentional? Also, a note about thread-safety?